### PR TITLE
Add references to iot TopicRule

### PR DESCRIPTION
**Pull request by mbbush**
### Description of your changes

Fixes #

I have:

- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-kafka.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-cloudwatchlogs.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-dynamodb.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-dynamodbv2.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-kafka.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-dynamodb.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-kinesis.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-firehose.yaml"

**Comment by mbbush** [comment]
I think I'm going to need some help from someone at Upbound to clean up some orphaned resources in the AWS account used by uptest. 

The first run of the topicrule-kafka example failed because I wasn't using ${data.aws_account_id}. I think that when the test timed out, it didn't evaluate the pre-deletion hook which makes sure to delete the iot TopicRuleDestination before deleting the IAM role it uses to clean up its ENIs. As a result, the created VPC still had resources left in it, and failed its deletion. 

Then when I fixed the example and reran, I got an error creating the vpc that there were too many VPCs in the account in us-east-2. IIRC the default VPC limit is 5 per region, so it would be great if aws could increase that for accounts used to test IaC. I think you can just submit a quota increase request.

TL;DR there's at least one VPC with a couple unused ENIs in it in us-east-2 that need to be deleted. 

@jeanduplessis is this something you could help with?

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-timestream.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-stepfunction.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-s3.yaml"

**Comment by jeanduplessis** [comment]
@mbbush not to worry about orphaned resources for failed runs. We have a process that goes and kills all resources in the AWS account we use for uptest on a regular cadence.

**Comment by mbbush** [comment]
/test-examples="examples/iot/topicrule-kafka.yaml"

**Comment by turkenf** [comment]
Are you still interested in taking this PR forward, @mbbush?
